### DISCUSSION TITLE
Centering LayoutTransformControl in ScrollViewer

### DISCUSSION THREAD

**Discussion by gebodal** [AvaloniaUI/Avalonia#15892] [Q&A] [answered]
Hello all,

I'm trying to recreate a WPF desktop application in Avalonia, and I've encountered what I assume is different default behaviour of the ScrollViewer control in the two platforms.

I have a control inside a ScrollViewer whose size is being adjusted by a LayoutTransform (for zooming purposes).

In WPF, this looks something like:

```
<ScrollViewer
		Name="CanvasView"
		Grid.Row="0"
		PreviewMouseWheel="OnCanvasPreviewMouseWheel"
		PreviewMouseLeftButtonDown="OnCanvasPreviewMouseLeftButtonDown"
		PreviewMouseLeftButtonUp="OnCanvasPreviewMouseLeftButtonUp"
		MouseMove="OnCanvasMouseMove"
		SizeChanged="OnCanvasSizeChanged"
		HorizontalScrollBarVisibility="Auto"
		VerticalScrollBarVisibility="Auto">
	<Canvas Name="CanvasContent">
		<Canvas.LayoutTransform>
			<TransformGroup x:Name="transform">
				<ScaleTransform x:Name="scaleTransform" ScaleX="1" ScaleY="1"/>
			</TransformGroup>
		</Canvas.LayoutTransform>
	</Canvas>
</ScrollViewer>
```

With the size and content of the Canvas being adjust in the code-behind, and the whole thing being inside a Grid.

In Avalonia, as far as I can tell, this becomes something like:

```
<ScrollViewer
		Name="CanvasView"
		Grid.Row="0"
		PointerPressed="OnCanvasPointerPressed"
		PointerReleased="OnCanvasPointerReleased"
		PointerMoved="OnCanvasPointerMoved"
		SizeChanged="OnCanvasSizeChanged"
		HorizontalScrollBarVisibility="Auto"
		VerticalScrollBarVisibility="Auto">
	<LayoutTransformControl Name="layoutTransformWrapper">
		<LayoutTransformControl.LayoutTransform>
			<ScaleTransform ScaleX="1" ScaleY="1"/>
		</LayoutTransformControl.LayoutTransform>
	</LayoutTransformControl>
</ScrollViewer>
```

And the LayoutTransformControl Child can be set to change the content. So far so good.

The problem is that when the content is smaller than the ScrollViewer, some odd behaviour happens in Avalonia where the content gets shunted outside of the viewport (towards the bottom right, if that matters). I have some code which controls the ScrollViewer offsets (to keep the location of the mouse pointer constant during scrolling), but the behaviour remains unchanged even with this code commented out (it doesn't actually seem to have an effect when the content is smaller than the viewport??).

In WPF, the default behaviour seemed to be that the content would be centred inside the ScrollViewer if it was smaller. I can't work out how to reproduce this behaviour in Avalonia. Do I need to wrap the LayoutTransformControl in another control or something? Should I be adjusting some property on the ScrollViewer?

Any suggestions would be greatly appreciated.

**Comment by Pratham95VP** [comment, marked as answer]
you can wrap the` LayoutTransformControl` in ` Grid` or a` StackPanel` and set the` HorizontalContentAlignment` and `VerticalContentAlignment` properties.

**Reply by gebodal** [reply]
Worked like a charm, thank you!
